### PR TITLE
add body receiver and sender

### PR DESCRIPTION
**Pull request by tannernelson**
This adds a receiver and sender stream type to the body.

It also has tests.

### CODE REVIEW

**Comment by tannernelson** [comment]
Needs swiftx/c7#6 to pass tests.

File: Tests/S4/BodyTests.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+        XCTAssert(data.bytes == bodyForBuffer.buffer.bytes, "Garbled buffer bytes")
+
+        (bodyForReceiver.receiver as? Drain)?.open()
+        let receiverDrain = Drain(bodyForReceiver.receiver) //must open because of reference semantics
+        XCTAssert(data.bytes == receiverDrain.data.bytes, "Garbled receiver bytes")
+
+
+        let senderDrain = Drain()
+        (bodyForReceiver.receiver as? Drain)?.open() //must open because of reference semantics
```
Comment (paulofaria) [inline Tests/S4/BodyTests.swift]: I think we should handle that inside `Drain`. remember we discussed `Drain(open: true)` or something like that?

File: Tests/S4/BodyTests.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        ]
+    }
+
+    let data: Data = [0x00, 0x01, 0x02, 0x03]
+
+    func testSender() {
+        let sender = Body.sender({ stream in
+            try stream.send(self.data)
+        })
+
```
Comment (tannernelson) [inline Tests/S4/BodyTests.swift]: Good point. I think I did Xcode autocomplete on that one. I'll update it.

File: Tests/S4/BodyTests.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        testBodyProperties(buffer)
+    }
+
+    private func testBodyProperties(body: Body) {
+        var bodyForBuffer = body
+        var bodyForReceiver = body
+        var bodyForSender = body
+
+        XCTAssert(data.bytes == bodyForBuffer.buffer.bytes, "Garbled buffer bytes")
+
```
Comment (tannernelson) [inline Tests/S4/BodyTests.swift]: I'll check. And change it

File: Tests/S4/BodyTests.swift (hunk 1 above)
Comment (tannernelson) [inline Tests/S4/BodyTests.swift]: Ah yeah. Good point. I'll update that in C7 then we should merge C7 first

**Pull request merged**